### PR TITLE
Initial k6 setup

### PR DESCRIPTION
**Pull request by kclark-scottlogic**
## Description

Initial setup PR for k6 to have first set of scenarios running properly.

Ticket #812 

## Screenshots

Where applicable, show screenshots of the visual changes.

## Notes

- Bullet point list with
- any notable code changes
- or explanations regarding this PR

## Concerns

- Bullet point list
- with any questions
- or concerns about this PR

## Checklist

Have you done the following?

- [x] Linked the relevant Issue
- [x] Added tests
- [x] Ensured the workflow steps are passing

### CODE REVIEW

**Review by pmarsh-scottlogic — APPROVED** [review]
nice nice nice

File: k6/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# K6 ReadMe
+
+## Introduction
+
+This readme is for how to install and us k6 for manual load testing
```
Comment (chriswilty) [inline k6/README.md:5]: typo: "use k6"

I think a good intro paragraph would explain what we are using k6 for, and why! So yes, generally performance testing, but also worth mentioning k6 is free and open-source 💚

File: k6/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+## Installation
+
+The link for installation of k6 can be found [here](https://grafana.com/docs/k6/latest/get-started/installation/). Follow the instructions on the link for your installer of choice
```
Comment (chriswilty) [inline k6/README.md:9]: Generally speaking, linking the word "here" is not good practice. The internet is full of it though 😢

Better to have the link text describe what it is linking to:

```
[Download and install k6](https://grafana.com/docs/k6/latest/get-started/installation/)
```

File: k6/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+## Running locally
+
+1. Make sure local backend is running
```
Comment (chriswilty) [inline k6/README.md:13]: Could link to the backend README here.

```
1. Start the server - [refer to the backend README](../backend/README.md)
```

File: k6/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+1. in a sepearate terminal run ```k6 run script.js``` (Where 'script' is the name of the script you want to run)
```
Comment (chriswilty) [inline k6/README.md:14]: Typo: `In a separate terminal`

Also, use single backticks for inline code, not triples.

File: k6/smoke.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import http from 'k6/http';
+import { check, sleep } from 'k6';
+
+export const options = {
+  vus: 3, // Key for Smoke test. Keep it at 2, 3, max 5 VUs
+  duration: '1m', // This can be shorter or just a few iterations
+};
+
+export default () => {
+  let data = { message: "hi", currentLevel: '3' };
```
Comment (chriswilty) [inline k6/smoke.js:10]: Use a `const`.

File: k6/smoke.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  const url = 'http://localhost:3001/openai/chat';
+
+  let response = http.post(url, JSON.stringify(data), {
+    headers: { 'Content-Type': 'application/json' },
+  });
+  check(response, {
+    'response code was 200': (response) => response.status == 200,
```
Comment (chriswilty) [inline k6/smoke.js:17]: Need triple-equals here.

**Review by chriswilty — COMMENTED** [review]
Tried it out locally, works a treat 👌 

Just a handful of suggestions

**Review by chriswilty — CHANGES_REQUESTED** [review]
Whoops, forgot to submit my review as a change request

File: k6/README.md [hunk 7]
```diff
@@ -2,13 +2,13 @@
 
 ## Introduction
 
-This readme is for how to install and us k6 for manual load testing
+Grafana k6 is an open-source load testing tool that makes performance testing easy and productive for engineering teams. k6 is free, developer-centric, and extensible. In this project we are using k6 for manual load testing. The following readMe will give basic instructions on where, what and how to install and run.
```
Comment (chriswilty) [inline k6/README.md:5]: ❤️

File: k6/README.md [hunk 8]
```diff
@@ -2,13 +2,13 @@
...
 
 ## Installation
 
-The link for installation of k6 can be found [here](https://grafana.com/docs/k6/latest/get-started/installation/). Follow the instructions on the link for your installer of choice
+The link for installation of k6 can be found [Download and install k6](https://grafana.com/docs/k6/latest/get-started/installation/). Follow the instructions on the link for your installer of choice
```
Comment (chriswilty) [inline k6/README.md:9]: Remove the leading "The link for installation of k6 can be found" text, as it makes no sense with that still there.

File: k6/README.md [hunk 9]
```diff
@@ -2,13 +2,13 @@
...
 
 ## Running locally
 
-1. Make sure local backend is running
-1. in a sepearate terminal run ```k6 run script.js``` (Where 'script' is the name of the script you want to run)
\ No newline at end of file
+1. Make sure local backend is running [refer to the backend README](../backend/README.md)
```
Comment (chriswilty) [inline k6/README.md:13]: comma after "running"

**Review by chriswilty — CHANGES_REQUESTED** [review]
Teeny tiny corrections in readme, but looking nice 👌

**Review by chriswilty — APPROVED** [review]
Sweet 👍

**Pull request merged**